_test/tests/inc/parser/handler_parse_highlight_options.test.php

```diff
@@ -0,0 +1,56 @@
+<?php
+
+class TestOfDoku_Handler_ParseHighlightOptions extends DokuWikiTest {
+
+    public function dataProvider() {
+        return [
+            ['', null],
+            ['something weird', null],
+            ['enable_line_numbers', ['enable_line_numbers' => true]],
+            ['enable_line_numbers=1', ['enable_line_numbers' => true]],
+            ['enable_line_numbers="1"', ['enable_line_numbers' => true]],
+            ['enable_line_numbers=0', ['enable_line_numbers' => false]],
+            ['enable_line_numbers="0"', ['enable_line_numbers' => false]],
+            ['enable_line_numbers=false', ['enable_line_numbers' => false]],
+            ['enable_line_numbers="false"', ['enable_line_numbers' => false]],
+            ['highlight_lines_extra', ['highlight_lines_extra' => [1]]],
+            ['highlight_lines_extra=17', ['highlight_lines_extra' => [17]]],
+            ['highlight_lines_extra=17,19', ['highlight_lines_extra' => [17, 19]]],
+            ['highlight_lines_extra="17,19"', ['highlight_lines_extra' => [17, 19]]],
+            ['highlight_lines_extra="17,19,17"', ['highlight_lines_extra' => [17, 19]]],
+            ['start_line_numbers_at', ['start_line_numbers_at' => 1]],
+            ['start_line_numbers_at=12', ['start_line_numbers_at' => 12]],
+            ['start_line_numbers_at="12"', ['start_line_numbers_at' => 12]],
+            ['enable_keyword_links', ['enable_keyword_links' => true]],
+            ['enable_keyword_links=1', ['enable_keyword_links' => true]],
+            ['enable_keyword_links="1"', ['enable_keyword_links' => true]],
+            ['enable_keyword_links=0', ['enable_keyword_links' => false]],
+            ['enable_keyword_links="0"', ['enable_keyword_links' => false]],
+            ['enable_keyword_links=false', ['enable_keyword_links' => false]],
+            ['enable_keyword_links="false"', ['enable_keyword_links' => false]],
+            [
+                'enable_line_numbers weird nothing highlight_lines_extra=17,19 start_line_numbers_at="12" enable_keyword_links=false',
+                [
+                    'enable_line_numbers' => true,
+                    'highlight_lines_extra' => [17, 19],
+                    'start_line_numbers_at' => 12,
+                    'enable_keyword_links' => false
+                ]
+            ],
+        ];
+    }
+
+    /**
+     * @dataProvider dataProvider
+     * @param string $input options to parse
+     * @param array|null $expect expected outcome
+     * @throws ReflectionException
+     */
+    public function testOptionParser($input, $expect) {
+        $h = new Doku_Handler();
```

Comment:
Is this ever executed? I assume there is some magic happening because of the ```@dataProvider dataProvider``` comment?

Reply:
Yes, there is magic :).

PHPUnit calls this function with every element in the array returned by the `dataprovider()` method.

You can see it for yourself by running `phpunit --filter TestOfDoku_Handler_ParseHighlightOptions` on the commandline in the `_test/` directory.

Documentation: https://phpunit.de/manual/6.5/en/writing-tests-for-phpunit.html#writing-tests-for-phpunit.data-providers

Reply:
Ok, thanks.